File: backend/spring/nlayer-architecture/minimal/src/main/java/com/apptemplate/api/dto/AuditLogDto.java
package com.apptemplate.api.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class AuditLogDto {

    private Long id;
    private String action;
    private String entityType;
    private String entityId;
    private Long userId;
    private String userName;
    private String details;
    private String ipAddress;
    private LocalDateTime createdAt;
}

File: backend/dotnet/feature-architecture/minimal/src/App.Template.Api/Features/AuditLogs/Dtos/AuditLogDto.cs
namespace App.Template.Api.Features.AuditLogs.Dtos;

public class AuditLogDto
{
    public long Id { get; set; }
    public string EntityType { get; set; } = string.Empty;
    public string EntityId { get; set; } = string.Empty;
    public string Action { get; set; } = string.Empty;
    public string? UserId { get; set; }
    public string? UserName { get; set; }
    public string? Details { get; set; }
    public string? IpAddress { get; set; }
    public DateTime CreatedAt { get; set; }
}
